head/src/test/kotlin/io/qalipsis/core/head/configuration/ZoneServiceTest.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.head.configuration
+
+import assertk.all
+import assertk.assertThat
+import assertk.assertions.hasSize
+import assertk.assertions.index
+import assertk.assertions.isEqualTo
+import assertk.assertions.prop
+import io.mockk.coEvery
+import io.mockk.coVerifyOrder
+import io.mockk.confirmVerified
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.spyk
+import io.qalipsis.core.head.jdbc.entity.ZoneEntity
+import io.qalipsis.core.head.jdbc.repository.ZoneRepository
+import io.qalipsis.core.head.model.Zone
+import io.qalipsis.test.coroutines.TestDispatcherProvider
+import io.qalipsis.test.mockk.WithMockk
+import kotlinx.coroutines.flow.flowOf
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import java.net.URL
+
+@WithMockk
+internal class ZoneServiceTest {
+
+    @JvmField
+    @RegisterExtension
+    val testDispatcherProvider = TestDispatcherProvider()
+
+    @RelaxedMockK
+    private lateinit var zoneRepository: ZoneRepository
+
+    private lateinit var zoneService: ZoneServiceImpl
```

Comment:
Use the annotations @InjectMocks and @Spyk on `zoneService` instead of creating the instance on the before all.